### PR TITLE
fix: synchronize the logic of clientLayout to defaultLayout, close #130

### LINKED ISSUE #130: default layout not processed when using sub folder without a layout
although a default layout is set when using a sub folder if a layout is not specified in the route tag the default layout is not processed

[`stackblitz`](https://github.com/cirolosapio/vite-plugin-vue-layouts-issue)

### PR DESCRIPTION
**Pull request by markthree**
@JohnCampionJr 
ClientLayout handled issue #130, so I synchronized the logic. If any changes ignore any cases, please let me know

### CODE REVIEW

**Comment by JohnCampionJr** [comment]
OK, Trying to confirm the edge cases here  All are fine except one.  

With vue-router, the `pages//second/index.vue` in the spa example, the layout is overridden to `second`.
With unplugin-vue-router, the second layout is inserted inside the default.   

I'm not sure there's any way to fix this due to the nature of unplugin-vue-router's structure.  Any ideas?

**Comment by JohnCampionJr** [comment]
OK, Trying to confirm the edge cases here All are fine except one.

With vue-router, the `pages/second/index.vue` in the spa example, the layout is overridden to second.
With unplugin-vue-router, the second layout is inserted inside the default.

I'm not sure there's any way to fix this due to the nature of unplugin-vue-router's structure. Any ideas?

**Comment by JohnCampionJr** [comment]
I did figure out that it can be fixed with a top-level `second.vue` with layout set to false

**Comment by markthree** [comment]
> I did figure out that it can be fixed with a top-level `second.vue` with layout set to false
nice

File: src/RouteLayout.ts [hunk 1]
```diff
@@ -17,14 +17,12 @@ export function setupLayouts(routes) {
...
         route.children = deepSetupLayout(route.children, false)
       }
-
-      if (!route.component) return route
       
       if (top && route.meta?.layout !== false) {
         return { 
           path: route.path,
           component: layouts[route.meta?.layout || '${options.defaultLayout}'],
-          children: route.path === '/' ? [route] : [{...route, path: ''}],
```
Comment (YunYouJun) [inline src/RouteLayout.ts:27]: Why change the code here? This is a special patch.

https://github.com/JohnCampionJr/vite-plugin-vue-layouts/pull/97

Now it's a problem again.

Comment (markthree) [reply src/RouteLayout.ts:27]: I apologize for overlooking this case.
I will revert this change while maintaining compatibility with solution #130

Comment (markthree) [reply src/RouteLayout.ts:27]: done → #137

**Pull request merged**